### ISSUE TITLE
NullPointerException when calling MetaImpl.snapshot() on a PostgreSQL schema with function based indexes

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10804]
When a PostgreSQL schema contains function based indexes such as:

```sql
CREATE INDEX i_author_name_upper ON t_author (upper(last_name), upper(first_name))
```

Then we get a `NullPointerException` on `DSLContext.meta().snapshot()` calls:

```
Exception in thread "main" java.lang.NullPointerException
	at org.jooq.impl.Snapshot$SnapshotTable.<init>(Snapshot.java:209)
	at org.jooq.impl.Snapshot$SnapshotSchema.<init>(Snapshot.java:142)
	at org.jooq.impl.Snapshot$SnapshotCatalog.<init>(Snapshot.java:109)
	at org.jooq.impl.Snapshot.getCatalogs0(Snapshot.java:94)
	at org.jooq.impl.AbstractMeta.initCatalogs(AbstractMeta.java:114)
	at org.jooq.impl.AbstractMeta.getCatalogs(AbstractMeta.java:106)
	at org.jooq.impl.Snapshot.<init>(Snapshot.java:79)
	at org.jooq.impl.AbstractMeta.snapshot(AbstractMeta.java:467)
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11256)

**Comment by lukaseder** [comment]
Similar issue: https://github.com/jOOQ/jOOQ/issues/11258
